src/dynamixel_hardware_interface.cpp

```diff
@@ -144,24 +179,41 @@ namespace dynamixel
     **/
     void DynamixelHardwareInterface::write_joints()
     {
-        std::vector<int> command_int;
-        command_int.resize(_dynamixel_ids.size());
 
-        for(unsigned int i=0; i<_dynamixel_ids.size(); i++)
-        {
-            command_int[i] = (int)(_joint_commands[i]*2048/M_PI) + 2048;
+        for (unsigned int i = 0; i < _dynamixel_servos.size(); i++) {
+            std::map<long long int, std::string>::iterator dynamixel_iterator = _dynamixel_map.find(_dynamixel_servos[i]->id());
+            if (dynamixel_iterator == _dynamixel_map.end())
+                continue;
+            // Sending commands only when needed
+            if (std::abs(_joint_commands[i] - _prev_commands[i]) < std::numeric_limits<double>::epsilon())
+                continue;
+            _prev_commands[i] = _joint_commands[i];
+            try {
+                dynamixel::StatusPacket<dynamixel::protocols::Protocol1> status;
+
+                double goal_pos = _joint_commands[i];
+
+                std::map<long long int, double>::iterator dynamixel_corrections_iterator = _dynamixel_corrections.find(_dynamixel_servos[i]->id());
+                if (dynamixel_corrections_iterator != _dynamixel_corrections.end()) {
```

Comment:
The important thing is to remove from the  `_dynamixel_servos` vector the
ones that are not in the map

2016-03-01 11:27 GMT+01:00 Konstantinos Chatzilygeroudis <
notifications@github.com>:

> In src/dynamixel_hardware_interface.cpp
> <https://github.com/resibots/dynamixel_control_hw/pull/2#discussion_r54547889>
> :
>
> > -            command_int[i] = (int)(_joint_commands[i]*2048/M_PI) + 2048;
> > +        for (unsigned int i = 0; i < _dynamixel_servos.size(); i++) {
> > +            std::map<long long int, std::string>::iterator dynamixel_iterator = _dynamixel_map.find(_dynamixel_servos[i]->id());
> > +            if (dynamixel_iterator == _dynamixel_map.end())
> > +                continue;
> > +            // Sending commands only when needed
> > +            if (std::abs(_joint_commands[i] - _prev_commands[i]) < std::numeric_limits<double>::epsilon())
> > +                continue;
> > +            _prev_commands[i] = _joint_commands[i];
> > +            try {
> > +                dynamixel::StatusPacket<dynamixel::protocols::Protocol1> status;
> > +
> > +                double goal_pos = _joint_commands[i];
> > +
> > +                std::map<long long int, double>::iterator dynamixel_corrections_iterator = _dynamixel_corrections.find(_dynamixel_servos[i]->id());
> > +                if (dynamixel_corrections_iterator != _dynamixel_corrections.end()) {
>
> It would be better at init to remove from the map the not-used
> dynamixels...
>
> I will fix this sometime today.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/resibots/dynamixel_control_hw/pull/2/files#r54547889>.
>